### PR TITLE
Fixed and improved logic related to recurrent events functionality and DropZone component

### LINKED ISSUE #598: Recurrent event functionality doesn't work properly
**Date Time**
24-05-2021 15:00

**Priority**
Medium

**Environment:**
 OS: Windows
 Browser: Google Chrome, Firefox
 Reproducible: always
 Build found: (e.g. tag)
 
**Pre-conditions**

Be authorized as a user

**Steps to Reproduce**

1. Click on 'Add event' button or go to "Draft" and select any available draft
2. Fill in the fields with all the necessary information about the event
3. Check the box 'Reccurent event'
3. Click on "Publish" button

**Expected result**
An event is added to the list of all events and becomes recurrent, and can also be found on the 'Recurrent events' page.

**Actual result**
An event is added to the list of all events, but doesn't become recurrent and doesn't appear on the 'Recurrent events' page.

### LINKED ISSUE #628: The old event image is loaded after being changed during editing and reopening the event edit page
**Date Time**
02-06-2021 12:00

**Priority**
Medium

**Environment:**
 OS: Windows 10 x64
 Browser: Google Chrome, Firefox
 Reproducible: always
 Build found: 20210601.7
 
**Pre-conditions**

Be authorized as a user
Have a draft of an event or own any published event with an uploaded image

**Steps to Reproduce**
1. Go to 'Draft' and select any available draft
    OR
    Go to "Home" and select any published event owned by the user, then click on 'Edit' button
2. Click on 'Clear' button to remove the image
3. Upload any other image that is different from the previous one
4. Click on 'Crop' button to crop the image
5. Click on 'Save' button
6. Go back to 'Draft'/event page
7. Select the same event
8. See the old image uploaded

**Expected result**
The new event image is loaded into the edit form after replacing the old one during editing and reopening the page

**Actual result**
The old event image is loaded into the edit form after being changed during editing and reopening the page

**Screenshots**
The old image on the preview card is loaded
Old image on the preview card
The new image is uploaded
New image
The new image on the preview card is loaded correctly
New image on the preview card
The wrong (old) image is uploaded to edit form
Old image

### LINKED ISSUE #630: The wrong shape of image is loaded after cropping the user's avatar
**Date Time**
03-06-2021 12:00

**Priority**
Medium

**Environment:**
 OS: Windows 10 x64
 Browser: Google Chrome, Firefox
 Reproducible: always
 Build found: 20210601.7
 
**Pre-conditions**

1. Go to https://eventsexpress-test.azurewebsites.net
2. Authorize as user

**Steps to Reproduce**
1. Click on 'Edit your profile' button
2. Click on 'Change avatar'
3. Crop the existing avatar or upload an image if it's not there and then crop it
4. See the wrong crop shape

**Expected result**
The round crop shape of image is loaded after cropping the user's avatar

**Actual result**
The rectangle crop shape of image is loaded after cropping the user's avatar

**Screenshots**
Wrong crop shape
Actual result
Desired crop shape
Expected result

### PR DESCRIPTION
**Pull request by alexanchez**
dev
## GitHub Project

* [Main GitHub Project ticket](https://github.com/EventsExpress/EventsExpress/projects/2)

## Code reviewers

- [x] @iuriikhrystiuk 

### Second Level Review

- [ ] @sand0 

## Summary of issue

The recurring events functionality doesn't work as expected.
The old event image is loaded after being changed during editing and reopening the event edit page.
The wrong shape of image is loaded after cropping the user's avatar.
Therefore, DropZone component needs changes to display images correctly.

## Summary of change

Fixed and improved flow related to event scheduling functionality.
Added logic for creating/editing/deleting the event schedule when editing an event draft. 
Fixed issues with the DropZone component and improved its logic. 
Fixed UI issues with displaying event image and user avatar. There are now different crop shapes for each type of image. 
Fixed and improved validation of the event form fields.
Removed unnecessary and redundant functionality.
Code refactoring has been performed.

## Testing approach

The unit tests have been written. Tested manually as well. 

## CHECK LIST
- [x]  PR is reviewed manually again (to make sure you have 100% ready code)
- [x]  I've checked new feature as logged in and logged out user if needed
- [x]  PR meets all conventions

### CODE REVIEW

File: EventsExpress.Core/Services/EventScheduleService.cs [hunk 1]
```diff
@@ -86,6 +91,25 @@ public async Task<Guid> Edit(EventScheduleDto eventScheduleDTO)
...
+        public async Task<Guid> Delete(Guid id)
+        {
+            var eventSchedule = Context.EventSchedules.Find(id);
+            if (eventSchedule == null)
+            {
+                throw new EventsExpressException("Not found");
+            }
+
+            var result = Delete(eventSchedule);
+            if (result.Id != id)
```
Comment (iuriikhrystiuk) [inline EventsExpress.Core/Services/EventScheduleService.cs]: This `if` makes no sense here. Please remove it.

File: EventsExpress/ClientApp/src/components/event/event-form.js [hunk 2]
```diff
@@ -110,6 +115,7 @@ class EventForm extends Component {
                             <Field
                                 name='dateFrom'
                                 label='From'
+                                disabled={disabledDate ? true: false}
```
Comment (iuriikhrystiuk) [inline EventsExpress/ClientApp/src/components/event/event-form.js]: It is already a boolean value, donot use ternary operator.

File: EventsExpress.Core/Services/EventScheduleService.cs (hunk 1 above)
Comment (alexanchez) [reply EventsExpress.Core/Services/EventScheduleService.cs]: Ok. I've already noticed and removed this when I was writing unit tests for this method. These changes will be delivered in the next commit when I finish writing all unit tests

File: EventsExpress/ClientApp/src/components/event/event-form.js (hunk 2 above)
Comment (alexanchez) [reply EventsExpress/ClientApp/src/components/event/event-form.js]: Ok. I'll fix this in the next commit

**Review by iuriikhrystiuk — APPROVED** [review]

**Pull request merged**